src/lib.rs

```diff
@@ -160,60 +118,102 @@ pub fn get_jvm_dyn_lib_file_name() -> &'static str {
 /// If `JAVA_HOME` env var is defined, the function returns it without any checks whether the var points to a valid directory or not.
 ///
 /// If `JAVA_HOME` is not defined, the function tries to locate it using the `java` executable.
-pub fn locate_java_home() -> errors::Result<String> {
+pub fn locate_java_home() -> Result<String> {
     match &env::var("JAVA_HOME") {
         Ok(s) if s.is_empty() => do_locate_java_home(),
         Ok(java_home_env_var) => Ok(java_home_env_var.clone()),
         Err(_) => do_locate_java_home(),
     }
 }
 
-fn do_locate_java_home() -> errors::Result<String> {
-    // Prepare the command depending on the host
-    let command_str = if is_windows() {
-        "where"
-    } else if is_macos() {
-        "/usr/libexec/java_home"
-    } else {
-        "which"
-    };
+#[cfg(target_os = "windows")]
+fn do_locate_java_home() -> Result<String> {
+    let output = Command::new("where")
+        .arg("java")
+        .output()
+        .map_err(|e| JavaLocatorError::new(format!("Failed to run command `where` ({e})")))?;
+
+    let java_exec_path = std::str::from_utf8(&output.stdout)?
+        // Windows will return multiple lines if there are multiple `java` in the PATH.
+        .lines()
+        // The first line is the one that would be run, so take just that line.
+        .next()
+        .unwrap()
+        .trim();
 
-    let mut command = Command::new(command_str);
+    if java_exec_path.is_empty() {
+        return Err(JavaLocatorError::new(
+            "Java is not installed or not in the system PATH".into(),
+        ));
+    }
+
+    let mut home_path = follow_symlinks(java_exec_path);
+
+    // Here we should have found ourselves in a directory like /usr/lib/jvm/java-8-oracle/jre/bin/java
```

Comment:
Please mention in the comment a valid Windows location

Reply:
that comment was taken from the original implementation on line 228 where it was in a codepath taken by both windows and linux, so I've not introduced this issue.
I'm not sure what the correct path would be for windows so I've just removed the comment. Feel free to provide me with a path you would rather me include here.